backend/compact-connect/lambdas/staff-user-pre-token/tests/test_main.py

```diff
@@ -19,15 +19,14 @@ def test_happy_path(self):
         # Create a DB record for this user's permissions
         self._table.put_item(
             Item={
-                'pk': sub,
-                'createdCompactJurisdiction': 'aslp/al',
+                'pk': f'USER#{sub}',
+                'sk': 'COMPACT#aslp',
+                'compact': 'aslp',
                 'permissions': {
-                    'aslp': {
-                        'actions': {'read'},
-                        'jurisdictions': {
-                            # should correspond to the 'aslp/write' and 'aslp/al.write' scopes
-                            'al': {'actions': {'write'}}
-                        }
+                    'actions': {'read'},
```

Comment:
The comment below says ' # should correspond to the 'aslp/write' and 'aslp/al.write' scopes' but this here is using the read action. Wouldn't that correspond to 'aslp/read' and not 'aslp/write'? Just want to make sure I understand the discrepancy between the comment and this line

Reply:
I'm not entirely sure what you're getting at. None of the scopes are *using* scopes. The read scope does come from line 26, and the two write scopes come from line 29. The comment is intended to point out that the one action corresponds to two scopes. Is my language just unclear here?

Reply:
Ok, so the parent scope 'aslp/write' is granted as a side effect of line 29, right?. This is not apparent to me as I look at the structure of the object. I had assumed that the 'aslp/write' permission was correlated to line 26, since that is the top level action for this particular user compact record. I was just trying to wrap my head around this where the scopes were coming from, and my intuitive guess was off. I assume we have the user scope/permissions model documented somewhere in the project which explains how these individual fields map to the final scopes the user has. No action item here, just trying to make sure I understand how the permissions model works. 😄

Reply:
Oh, gatcha. Yeah, the `aslp/write` coming from the jurisdiction action isn't really a side-effect so much as a direct consequence of that action, which needs to be represented by two scopes (course and fine -grained). [Here is the design documentation](https://github.com/InspiringApps/JCC-CompactConnect/tree/feat/staff-user-management-api/backend/compact-connect/docs/design#user-architecture), for what it's worth.